Fix/gosec linters

cwarnerdev opened the pull request:
More linter fixes. Mostly overflow checks in `packages/util/rwutil` and some `todo`s on VM things to return to later.

lunfardo314 commented on contracts/wasm/fairroulette/client/events.ts:
```diff
@@ -0,0 +1,114 @@
+type Hname = string;
+type Int16 = number;
+type Int32 = number;
+type Int64 = bigint;
```
Bigint probably has to be big.Int. It is coming with new VM and native tokens so maybe it makes sense  to reserve the name for the real thing

BugFreeSoftware replied:
Yeah I only just started on the client side code. Haven't even set up a debugging system for it yet.
But this is the JS bigint type. It's the only type that can represent int64 since number in JS is always stored as float. If we can do bigint with the VM in the future then it will probably be named Bigint or BigInt in WasmLib to be consistent with the rest of the naming.

cwarnerdev committed:
Add nolint comments for timestamp-related code

Added `//nolint: gosec` comments to suppress linter warnings on timestamp handling. A TODO was also added to review if changes can be made without breaking functionality.

cwarnerdev committed:
Use safecast for safe integer conversions in blocklog

Replaced direct type casting with safecast to ensure safe conversion of integers to uint16. Added error handling where necessary to handle potential conversion failures. This improves robustness and prevents possible runtime errors caused by unsafe typecasting.

cwarnerdev committed:
Add nolint comments to bypass gosec overflow warnings

The overflow checks are already handled above the return statements, so these nolint directives prevent unnecessary linter warnings. This ensures the code remains clean and adheres to its intended logic without redundant alerts.

cwarnerdev committed:
Use safecast for type-safe conversions in Reader methods

Updated `ReadDuration`, `ReadInt8`, `ReadInt16`, `ReadInt32`, and `ReadInt64` methods to use `safecast.Convert` for safer type casting. This change improves error handling by checking for conversion issues and setting `rr.Err` appropriately. It ensures more robust processing of data when reading various integer types.

cwarnerdev committed:
Refactor Writer class to use safecast for type conversions

Replaced direct type casts with safecast to ensure safer conversions and handle potential errors. Updated methods like WriteDuration, WriteInt8, WriteInt16, WriteInt32, and WriteInt64 to use safecast logic and store any errors encountered during conversion in the Writer's error field. This improves robustness and error handling for type conversions.
